src/crypto/CHIPCryptoPAL.cpp

```diff
@@ -608,5 +608,54 @@ CHIP_ERROR EcdsaAsn1SignatureToRaw(size_t fe_length_bytes, const ByteSpan & asn1
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR GenerateCompressedFabricId(const Crypto::P256PublicKey & root_public_key, uint64_t fabric_id,
+                                      MutableByteSpan & out_compressed_fabric_id)
+{
+    constexpr uint8_t kUncompressedPointMarker = 0x04;
+    if ((root_public_key.Length() != kP256_PublicKey_Length) || (root_public_key.ConstBytes()[0] != kUncompressedPointMarker))
+    {
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+    else if (out_compressed_fabric_id.size() < kCompressedFabricIdentifierSize)
```

Comment:
q: don't we want a uniform code stile of using `VerifyOrReturnError` macros?

Reply:
VerofyOrReturnError for long conditions just makes it less readable and more mental burden. I can change it if it's truly necessary.

Reply:
Done.